tuf/repository_tool.py

```diff
@@ -2290,8 +2292,8 @@ def delegate(self, rolename, public_keys, list_of_targets, threshold=1,
                 'terminating': terminating,
                 'paths': list(roleinfo['paths'].keys())}
 
-    if restricted_paths is not None:
-      roleinfo['paths'] = relative_restricted_paths
+    if paths is not None:
```

Comment:
`paths` can't be `None`, which means that `roleinfo['paths']` always gets `relative_paths` assigned, which means there is no reason to initialize it with `list(roleinfo['paths'].keys()`.
Also, is it expected that `roleinfo['paths']` can be a dictionary or a list?